src/app/components/forms/dynamic-form/terra-dynamic-form.component.ts

```diff
@@ -121,14 +122,24 @@ export class TerraDynamicFormComponent implements OnInit, OnChanges
     {
         if(!isNullOrUndefined(this.inputFormFunctions.onValueChangedCallback))
         {
-            this._formFieldControlService
+            let debounce:number = 1000;
+            if ( !isNullOrUndefined(this.inputFormFunctions.valueChangeDebounce) )
+            {
+                debounce = this.inputFormFunctions.valueChangeDebounce;
+            }
+            let stream:Observable<any> = this._formFieldControlService
                 .dynamicFormGroup
-                .valueChanges
-                .debounceTime(1000)
-                .subscribe((value:any) =>
-                {
-                    this.inputFormFunctions.onValueChangedCallback(value);
-                });
+                .valueChanges;
+
+            if ( debounce > 0 )
+            {
+                stream = stream.debounceTime(debounce);
+            }
+
+            stream.subscribe((value:any) =>
+            {
+                this.inputFormFunctions.onValueChangedCallback(value);
```

Comment:
Yes it is. Depending on the use case you might want to get the changes immediately. In my case, in need to get the changes directly because the bound model might change very quickly. In this case an get the change handler called after my bound model has changed and i will assign changed values to the wrong model.

Reply:
Ok